### PR TITLE
Fix BlockPayload conversion in SDK adapter

### PR DESCRIPTION
**Pull request by bartolomej**
Closes: https://github.com/onflow/flow-emulator/issues/703

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: adapters/sdk.go [hunk 1]
```diff
@@ -208,11 +208,33 @@ func (b *SDKAdapter) GetBlockByID(
...
+			ExecutionReceiptID: sdk.Identifier(seal.ResultID),
+		})
+	}
+
+	collectionGuarantees := make([]*sdk.CollectionGuarantee, 0)
+	for _, guarantee := range payload.Guarantees {
+		collectionGuarantees = append(collectionGuarantees, &sdk.CollectionGuarantee{
+			CollectionID: sdk.Identifier(guarantee.CollectionID),
+		})
+	}
```
Comment (turbolent) [inline adapters/sdk.go:230]: Pre-allocate, if needed:

```suggestion
	var seals []*sdk.BlockSeal
	sealCount := len(payload.Seals)
	if sealCount > 0 {
		seals = make([]*sdk.BlockSeal, 0, sealCount)
		for _, seal := range payload.Seals {
			seals = append(seals, &sdk.BlockSeal{
				BlockID:            sdk.Identifier(seal.BlockID),
				ExecutionReceiptID: sdk.Identifier(seal.ResultID),
			})
		}
	}

	var collectionGuarantees []*sdk.CollectionGuarantee
	guaranteesCount := len(payload.Guarantees)
	if guaranteesCount > 0 {
		collectionGuarantees = make([]*sdk.CollectionGuarantee, 0, guaranteesCount)
		for _, guarantee := range payload.Guarantees {
			collectionGuarantees = append(collectionGuarantees, &sdk.CollectionGuarantee{
				CollectionID: sdk.Identifier(guarantee.CollectionID),
			})
		}
	}
```

**Review by turbolent — APPROVED** [review]
Nice, thank you!

**Comment by turbolent** [comment]
Could you please add a/some test(s)?

File: adapters/sdk.go (hunk 1 above)
Comment (bartolomej) [reply adapters/sdk.go]: Fair point, fixed in https://github.com/onflow/flow-emulator/pull/704/commits/5fc70a20f052256cad4476710dc0ff6093c39a4a

**Comment by bartolomej** [comment]
> Could you please add a/some test(s)?

Yep, good idea! I actually just forgot to update them as they were failing. Fixed in https://github.com/onflow/flow-emulator/pull/704/commits/55388e7d2a6359de55075a04f36399888909dc07

File: adapters/sdk.go [hunk 2]
```diff
@@ -208,11 +208,41 @@ func (b *SDKAdapter) GetBlockByID(
...
+	sealCount := len(payload.Seals)
+	if sealCount > 0 {
+		seals = make([]*sdk.BlockSeal, 0, sealCount)
+		for _, seal := range payload.Seals {
+			seals = append(seals, &sdk.BlockSeal{
+				BlockID:            sdk.Identifier(seal.BlockID),
+				ExecutionReceiptID: sdk.Identifier(seal.ResultID),
+			})
+		}
+	}
```
Comment (sideninja) [inline adapters/sdk.go:227]: a bit cleaner (can do same bellow).
```suggestion
	seals := make(make([]*sdk.BlockSeal, len(payload.Seals))
	for i, seal := range payload.Seals {
			seals[i] = &sdk.BlockSeal{
				BlockID:            sdk.Identifier(seal.BlockID),
				ExecutionReceiptID: sdk.Identifier(seal.ResultID),
			}
		}
	}
```

Comment (turbolent) [reply adapters/sdk.go:227]: See https://github.com/onflow/flow-emulator/pull/704#discussion_r1667773134. The current code doesn't allocate unless necessary

Comment (bartolomej) [reply adapters/sdk.go:227]: I believe if capacity or length is set to 0, nothing should be allocated either: 

https://github.com/golang/go/blob/6d89b38ed86e0bfa0ddaba08dc4071e6bb300eea/src/runtime/slice.go#L116

https://github.com/golang/go/blob/6d89b38ed86e0bfa0ddaba08dc4071e6bb300eea/src/runtime/malloc.go#L988-L990

But yeah, It's probably still clearer to explicitly check for length in the application code.

**Pull request merged**